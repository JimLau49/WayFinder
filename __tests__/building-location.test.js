import React from "react";
import BuildingLocation from "../src/components/building-location/building-location.component";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("BuildingLocation component", () => {
  it("BuildingLocation: snapshot renders correctly", () => {
    const tree = renderer.create(<BuildingLocation />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("onBuildingLocationPress: check if button is pressed", () => {
    const mockOnBuildingLocationPress = jest.fn();
    const wrapper = shallow(
      <BuildingLocation onBuildingLocationPress={mockOnBuildingLocationPress} />
    );
    const button = wrapper.find("View TouchableOpacity");
    button.simulate("press");
    expect(mockOnBuildingLocationPress).toHaveBeenCalledTimes(1);
  });
});
