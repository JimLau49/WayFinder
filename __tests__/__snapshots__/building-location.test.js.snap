// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingLocation component BuildingLocation: snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "bottom": 110,
      "position": "absolute",
      "right": 20,
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "rgba(0,0,0,0.2)",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "width": 60,
        }
      }
    >
      <Text />
    </View>
  </RNGestureHandlerButton>
</View>
`;
